src/pipe_log.rs

```diff
@@ -74,6 +75,49 @@ impl FileBlockHandle {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq)]
+enum Content {
+    Checksum(u32),
+}
+
+#[derive(Debug, Clone)]
+pub struct Signature {
```

Comment:
I prefer a `FileContext` that contains both file id and format version. And log batch provides methods `prepare_write(ctx: &FileContext)`, `LogItemBatch::decode(.., ctx: &FileContext)`.